### PR TITLE
Generator: handle negative requirements a little better

### PR DESCRIPTION
**Pull request by Miepee**
Ran 2k of am2r+prime+echoes, couldnt see any big regressions so far.

prime and dread probably would benefit the most from this

### CODE REVIEW

File: randovania/generator/filler/pickup_list.py [hunk 1]
```diff
@@ -46,7 +46,9 @@ def _unsatisfied_item_requirements_in_list(
     damage = []
 
     for individual in alternative.values():
-        if individual.negate:
+        # We should only skip negative requirements, if we can not satisfy them anymore, as that means
+        # we have lost the chance to do so
+        if individual.negate and not individual.satisfied(state.resoures, state.energy, state.resource_database):
```
Comment (ThanatosGit) [inline randovania/generator/filler/pickup_list.py:51]: This looks not correct. `resoures` is a typo and maybe your result will be completly different if you use `resources`

**Review by ThanatosGit — DISMISSED** [review]

File: randovania/generator/filler/pickup_list.py (hunk 1 above)
Comment (Miepee) [reply randovania/generator/filler/pickup_list.py]: Whoops, no idea how that made it in.

Comment (ThanatosGit) [reply randovania/generator/filler/pickup_list.py]: On a second thought: This is python. Shouldn't it throw an `AttributeError` if this was ever called?

File: randovania/generator/filler/pickup_list.py [hunk 2]
```diff
@@ -46,7 +46,9 @@ def _unsatisfied_item_requirements_in_list(
     damage = []
 
     for individual in alternative.values():
-        if individual.negate:
+        # We should only skip negative requirements, if we can not satisfy them anymore, as that means
+        # we have lost the chance to do so
+        if individual.negate and not individual.satisfied(state.resources, state.energy, state.resource_database):
```
Comment (henriquegemignani) [inline randovania/generator/filler/pickup_list.py:51]: You could move the existing check to after the existing satisfied check. Could just add an ` or individual.negate`
Damage requirements are never negated (there's an assert enforcing that).

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

**Pull request merged**